src/stirling/source_connectors/socket_tracer/uprobe_manager.cc

```diff
@@ -280,53 +281,78 @@ StatusOr<std::vector<std::filesystem::path>> FindHostPathForPIDLibs(
                                 HostPathForPIDPathSearchType::kSearchTypeEndsWith);
 }
 
+// ssl_lib_matches allows customizing the search of shared object files
+// that need to be traced with the SSL_write and SSL_read uprobes.
+// In dynamically linked cases, it's likely that there are two
+// shared libraries (libssl and libcrypto). While statically linked
+// cases are contained within the same binary. ssl_library_match
+// allows for handling both cases. If there are two libraries present
+// in the std::vector, the former is assumed to be libssl and the
+// latter libcrypto.
+using ssl_lib_matcher =
+    std::vector<std::tuple<std::vector<std::string_view>, HostPathForPIDPathSearchType>>;
```

Comment:
A struct `LibSSLMatcher` could be easier to follow:
```
struct SSLLibMatcher {
  struct VanillaSSLLibs {
    std::string_view libssl;
    std::string_view libcrypto;
  };
  std::variant<VanillaSSLLibs, std::string_view> ssl_lib_patterns;
  HostPathForPIDPathSearchType search_type;
};
```

Reply:
Thanks for the suggestion. I like this approach better than the original method and my latest update uses this.